### ISSUE TITLE
DNF5 repoclosure - performance, rich dependencies, weak dependencies

### ISSUE THREAD

**Issue by j-mracek** [rpm-software-management/dnf5#1073]
DNF5 implementation is similar to DNF4, but it is significantly slower. The code inherited problems from DNF4 - rich deps evaluation and it does not test weak dependencies.

- [ ] investigate performance problem
- [ ] Investigate whether the code cannot be replaced by libsolv transaction with SOLVER_VERIFY flag.

**Comment by quantum-quirks** [comment]
Still horribly slow even today. I wonder what was the reason for dnf5 if it inherited all the problems of dnf4

**Comment by evan-goode** [comment]
@kontura repoclosure is quite fast now with https://github.com/rpm-software-management/dnf5/pull/2375, what's left here? Testing of weak dependencies?

**Comment by kontura** [comment]
Yes I suppose the weak deps and if we want to we could investigate the approach with libsolv transaction and SOLVER_VERIFY as mentioned by j-mracek.
